src/test/java/org/broadinstitute/hellbender/utils/nio/ExtremeReadsTest.java

```diff
@@ -0,0 +1,106 @@
+
+    static String fname = GCS_GATK_TEST_RESOURCES + "large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam";
+
+    static int THREAD_COUNT = 1000;
+    static int CHANNELS_PER_THREAD = 1000;
+    static volatile int errors = 0;
+
+    private static class Runner implements Runnable {
+        /**
+         * Read a bunch of bytes. Part of the manyParallelReads test.
+         */
+        @Override
+        public void run() {
+            try {
+                Path path = IOUtils.getPath(ExtremeReadsTest.fname);
+                ArrayList<SeekableByteChannel> chans = new ArrayList<SeekableByteChannel>();
+                for (int i=0; i<CHANNELS_PER_THREAD; i++) {
+                    SeekableByteChannelPrefetcher chan = new SeekableByteChannelPrefetcher(
+                        Files.newByteChannel(path), 2 * 1024 * 1024);
+                    // skip the first half
+                    chan.position(chan.position()/2);
+                    chans.add(chan);
+                }
+                long size = chans.get(0).size();
+                ByteBuffer buf = ByteBuffer.allocate(1024*1024 - 5);
+                while (!chans.isEmpty()) {
+                    SeekableByteChannel chan = chans.remove(0);
+                    buf.clear();
+                    int read = chan.read(buf);
+                    if (read>=0) {
+                        chans.add(chan);
+                        continue;
+                    }
+                    // EOF
+                    long position = chan.position();
+                    if (size != position) {
+                        System.out.println("Done at wrong position! " + position + " != " + size);
+                        ExtremeReadsTest.errors++;
+                    }
+                }
+            } catch (Exception x) {
+                ExtremeReadsTest.errors++;
+                System.out.println("Caught: " + x.getMessage());
+                x.printStackTrace();
+                System.out.println();
+            }
+        }
+    }
+
+    /**
+     * This test takes about a half hour and reads a fair amount of data.
+     * It definitely shouldn't be part of the normal test suite (that's why it's disabled)
+     * but it's kept here so we can manually run it should we need to investigate mysterious
+     * disconnects again.
+     **/
+    @Test(groups={"bucket"}, enabled=false)
+    public void manyParallelReads() throws InterruptedException {
+        final ExecutorService executor = Executors.newFixedThreadPool(THREAD_COUNT,
+            new ThreadFactory() {
+                public Thread newThread(Runnable r) {
+                    Thread t = Executors.defaultThreadFactory().newThread(r);
+                    t.setDaemon(true);
+                    return t;
+                }
+            });
+        Stopwatch sw = Stopwatch.createStarted();
+        errors = 0;
+        final Runner runner = new Runner();
+        for (int i=0; i<THREAD_COUNT; i++) {
+            executor.execute(runner);
```

Comment:
Pass in a separate `Runner` instance on each iteration.

Reply:
OK, but why? Runner is stateless.